File: queen-csharp/ChessBoard.cs
using System;
using System.Collections.Generic;

namespace QueenApplication
{
    class ChessBoard
    {
        public static readonly int SIZE = 8;

        private String[,] table;

        public ChessBoard()
        {
            InitBoard();
        }

        private void InitBoard()
        {
            table = new String[SIZE, SIZE];

            for (int i = 0; i < SIZE; i++)
            {
                for (int j = 0; j < SIZE; j++)
                {
                    table[i, j] = Piece.EMPTY.Value();
                }
            }
        }

        /**
         * setting chess pieces on board
         */
        public void SetPieces()
        {
            //black pieces
            table[0, 0] = Piece.BLACK_ROOK.Value();
            table[2, 2] = Piece.BLACK_HORSE.Value();
            table[0, 2] = Piece.BLACK_BISHOP.Value();
            table[0, 3] = Piece.BLACK_QUEEN.Value();
            table[0, 4] = Piece.BLACK_KING.Value();
            table[0, 5] = Piece.BLACK_BISHOP.Value();
            table[0, 6] = Piece.BLACK_HORSE.Value();
            table[0, 7] = Piece.BLACK_ROOK.Value();

            table[1, 0] = Piece.BLACK_PAWN.Value();
            table[1, 1] = Piece.BLACK_PAWN.Value();
            table[1, 2] = Piece.BLACK_PAWN.Value();
            table[1, 3] = Piece.BLACK_PAWN.Value();
            table[1, 4] = Piece.BLACK_PAWN.Value();
            table[1, 5] = Piece.BLACK_PAWN.Value();
            table[1, 6] = Piece.BLACK_PAWN.Value();
            table[1, 7] = Piece.BLACK_PAWN.Value();

            //white pieces
            table[7, 0] = Piece.WHITE_ROOK.Value();
            table[7, 1] = Piece.WHITE_HORSE.Value();
            table[7, 2] = Piece.WHITE_BISHOP.Value();
            table[5, 3] = Piece.WHITE_QUEEN.Value();
            table[7, 4] = Piece.WHITE_KING.Value();
            table[7, 5] = Piece.WHITE_BISHOP.Value();
            table[7, 6] = Piece.WHITE_HORSE.Value();
            table[7, 7] = Piece.WHITE_ROOK.Value();

            table[6, 0] = Piece.WHITE_PAWN.Value();
            table[6, 1] = Piece.WHITE_PAWN.Value();
            table[6, 2] = Piece.WHITE_PAWN.Value();
            table[6, 3] = Piece.WHITE_PAWN.Value();
            table[4, 4] = Piece.WHITE_PAWN.Value();
            table[6, 5] = Piece.WHITE_PAWN.Value();
            table[6, 6] = Piece.WHITE_PAWN.Value();
            table[6, 7] = Piece.WHITE_PAWN.Value();
        }

        public void PrintBoard()
        {
            Console.Write("\t");
            for (int c = 97; c < 105; c++)
            {
                Console.Write((char)c + "  ");
            }
            Console.WriteLine("\n\t-----------------------");

            for (int i = 0; i < SIZE; i++)
            {
                Console.Write(SIZE - i + "|\t");

                for (int j = 0; j < SIZE; j++)
                {
                    Console.Write(table[i, j] + " ");
                }

                Console.WriteLine();
            }
        }

        public List<String> GetAllPossibleMoves(Piece piece)
        {

            for (int i = 0; i < SIZE; i++)
            {
                for (int j = 0; j < SIZE; j++)
                {

                    if (table[i, j] == piece.Value())
                    {
                        if (piece.GetColor() == Piece.Color.WHITE)
                        {
                            SetBlackAttackZones();
                        }
                        if (piece.GetColor() == Piece.Color.BLACK)
                        {
                            SetWhiteAttackZones();
                        }

                        List<Coord> coordList = Movement.GetQueenAllPossibleMoves(new int[] { i, j }, table, piece);

                        RemoveAttackZones();

                        return ConvertToReadableCoord(coordList);
                    }
                }
            }

            return null;
        }

        private void SetBlackAttackZones()
        {

            for (int i = 0; i < SIZE; i++)
            {
                for (int j = 0; j < SIZE; j++)
                {
                    if (table[i, j] == Piece.BLACK_ROOK.Value())
                    {
                        Movement.SetRookAttackZone(new int[] { i, j }, table, Piece.BLACK_ROOK);
                    }

                    if (table[i, j] == Piece.BLACK_HORSE.Value())
                    {
                        Movement.SetHorseAttackZone(new int[] { i, j }, table, Piece.BLACK_HORSE);
                    }

                    if (table[i, j] == Piece.BLACK_BISHOP.Value())
                    {
                        Movement.SetBishopAttackZone(new int[] { i, j }, table, Piece.BLACK_BISHOP);
                    }

                    if (table[i, j] == Piece.BLACK_QUEEN.Value())
                    {
                        Movement.SetQueenAttackZone(new int[] { i, j }, table, Piece.BLACK_QUEEN);
                    }

                    if (table[i, j] == Piece.BLACK_KING.Value())
                    {
                        Movement.SetKingAttackZone(new int[] { i, j }, table, Piece.BLACK_KING);
                    }

                    if (table[i, j] == Piece.BLACK_PAWN.Value())
                    {
                        Movement.SetPawnAttackZone(new int[] { i, j }, table, Piece.BLACK_PAWN);
                    }
                }
            }

        }

        private void SetWhiteAttackZones()
        {
            //need to implement
        }


        private void RemoveAttackZones()
        {
            for (int i = 0; i < SIZE; i++)
            {
                for (int j = 0; j < SIZE; j++)
                {
                    if (table[i, j] == Piece.BLACK_ATTACK_ZONE.Value() || table[i, j] == Piece.WHITE_ATTACK_ZONE.Value())
                    {
                        table[i, j] = Piece.EMPTY.Value();
                    }
                }
            }
        }

        private static List<String> ConvertToReadableCoord(List<Coord> coordList)
        {
            List<String> result = new List<String>();

            foreach (Coord c in coordList)
            {
                String yy = Utils.GetLetterName(c.y);
                String xx = Math.Abs(c.x - SIZE).ToString();

                result.Add("( " + yy + "," + xx + " )");
            }

            return result;
        }


    }
}

File: queen/src/main/java/com/scatum/queen/ChessBoard.java
package com.scatum.queen;

import java.util.ArrayList;

import static com.scatum.queen.Piece.*;

class ChessBoard {

    static final int size = 8;

    private String[][] table;

    ChessBoard() {
        initBoard();
    }

    private void initBoard() {
        table = new String[size][size];

        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                table[i][j] = EMPTY.value();
            }
        }
    }

    void printBoard() {
        System.out.print("\t");
        for (int c = 97; c < 105; c++) {
            System.out.print((char) c + "  ");
        }
        System.out.println("\n\t-----------------------");

        for (int i = 0; i < size; i++) {

            System.out.print(size - i + "|\t");

            for (int j = 0; j < size; j++) {
                System.out.print(table[i][j] + " ");
            }
            System.out.println();
        }
    }

    /**
     * setting chess pieces on board
     */
    void setPieces() {

        //black pieces
        table[0][0] = BLACK_ROOK.value();
        table[2][2] = BLACK_HORSE.value();
        table[0][2] = BLACK_BISHOP.value();
        table[0][3] = BLACK_QUEEN.value();
        table[0][4] = BLACK_KING.value();
        table[0][5] = BLACK_BISHOP.value();
        table[0][6] = BLACK_HORSE.value();
        table[0][7] = BLACK_ROOK.value();

        table[1][0] = BLACK_PAWN.value();
        table[1][1] = BLACK_PAWN.value();
        table[1][2] = BLACK_PAWN.value();
        table[1][3] = BLACK_PAWN.value();
        table[1][4] = BLACK_PAWN.value();
        table[1][5] = BLACK_PAWN.value();
        table[1][6] = BLACK_PAWN.value();
        table[1][7] = BLACK_PAWN.value();

        //white pieces
        table[7][0] = WHITE_ROOK.value();
        table[7][1] = WHITE_HORSE.value();
        table[7][2] = WHITE_BISHOP.value();
        table[5][5] = WHITE_QUEEN.value();
        table[7][4] = WHITE_KING.value();
        table[7][5] = WHITE_BISHOP.value();
        table[7][6] = WHITE_HORSE.value();
        table[7][7] = WHITE_ROOK.value();

        table[6][0] = WHITE_PAWN.value();
        table[6][1] = WHITE_PAWN.value();
        table[6][2] = WHITE_PAWN.value();
        table[6][3] = WHITE_PAWN.value();
        table[4][4] = WHITE_PAWN.value();
        table[6][5] = WHITE_PAWN.value();
        table[6][6] = WHITE_PAWN.value();
        table[6][7] = WHITE_PAWN.value();
    }

    ArrayList<String> getAllPossibleMoves(Piece piece) {

        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {

                if (table[i][j] == piece.value()) {
                    if (piece.color() == Color.WHITE) {
                        setBlackAttackZones();
                    }
                    if (piece.color() == Color.BLACK) {
                        setWhiteAttackZones();
                    }

                    ArrayList<Coord> coordList = PieceMove.getQueenAllPossibleMoves(new int[]{i, j}, table, piece);

                    removeAttackZones();

                    return convertToReadableCoord(coordList);
                }
            }
        }

        return null;
    }

    private static ArrayList<String> convertToReadableCoord(ArrayList<Coord> coords) {
        ArrayList<String> result = new ArrayList<>();

        for (Coord c : coords) {
            String yy = Letter.getName(c.y);
            String xx = Integer.toString(Math.abs(c.x - size));

            result.add("( " + yy + "," + xx + " )");
        }

        return result;
    }

    private void setBlackAttackZones() {

        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                if (table[i][j] == BLACK_ROOK.value()) {
                    PieceMove.setRookAttackZone(new int[]{i, j}, table, BLACK_ROOK);
                }

                if (table[i][j] == BLACK_HORSE.value()) {
                    PieceMove.setHorseAttackZone(new int[]{i, j}, table, BLACK_HORSE);
                }

                if (table[i][j] == BLACK_BISHOP.value()) {
                    PieceMove.setBishopAttackZone(new int[]{i, j}, table, BLACK_BISHOP);
                }

                if (table[i][j] == BLACK_QUEEN.value()) {
                    PieceMove.setQueenAttackZone(new int[]{i, j}, table, BLACK_QUEEN);
                }

                if (table[i][j] == BLACK_KING.value()) {
                    PieceMove.setKingAttackZone(new int[]{i, j}, table, BLACK_KING);
                }

                if (table[i][j] == BLACK_PAWN.value()) {
                    PieceMove.setPawnAttackZone(new int[]{i, j}, table, BLACK_PAWN);
                }
            }
        }

    }

    private void setWhiteAttackZones() {
        //need to implement
    }

    private void removeAttackZones() {
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                if (table[i][j] == BLACK_ATTACK_ZONE.value() || table[i][j] == WHITE_ATTACK_ZONE.value()) {
                    table[i][j] = EMPTY.value();
                }
            }
        }
    }
}
